packages/notification-services-controller/package.json

```diff
@@ -68,7 +68,7 @@
   },
   "peerDependencies": {
     "@metamask/keyring-controller": "^17.0.0",
-    "@metamask/profile-sync-controller": "^0.2.0"
+    "@metamask/profile-sync-controller": "^0.2.1"
```

Comment:
This is a breaking change. This should have resulted in a minor version bump (since this package is pre-v1), but it looks like it wasn't even referenced in the changelog

Reply:
True, this should've been a minor instead of a patch.

How critical is this to revert?
Our only client consuming this is mobile.

Reply:
Is this diff necessary? `"^0.2.0"` should be compatible with `"0.2.1"` so there is no need to update to `"^0.2.1"` unless we're specifically trying to prevent `"0.2.0"` from being used. 

If avoiding `"0.2.0"` is the intention, that's a breaking change that probably should be communicated in the v182.0.0 release notes, not just the `@metamask/notification-services-controller@0.2.1` changelog. This will only be possible through a revert, not a changelog backport.

On a more general note, we should probably be using the tilde (~) operator e.g. `"~0.2.0"` instead of the carat (^) for versioning pre-v1 packages. With the carat operator, `"^0.x.y"` will all resolve as the same version, which makes it meaningless and contrary to our intentions here. With tilde, `"~0.2.0"` and `"~0.2.1"` will both resolve to `"0.2.0"`, while `"~0.3.1"` will resolve to `"0.3.0"`.

Reply:
Ah yes I deleted that part because I realized it contradicts the entire point of my comment😂 (e.g. `^0.3.1` and `~0.3.1` both shouldn't resolve to `0.3.0`). 

But I wasn't aware of the subtlety around leading zeros and missing patch or minor values. Guess the carat should work here as is. TIL!